tiledb/sm/rest/rest_client.cc

```diff
@@ -312,7 +323,14 @@ size_t RestClient::post_data_write_cb(
     Buffer* const scratch,
     Query* query,
     std::unordered_map<std::string, serialization::QueryBufferCopyState>*
-        copy_state) {
+        copy_state,
+    bool* const user_buffers_overflowed) {
+  // If 'user_buffers_overflowed' has been set, we should prevent all future
+  // response processing.
+  if (*user_buffers_overflowed) {
```

Comment:
Do you need a nullptr check here?

Reply:
It wouldn't hurt, but not necessarily. This is used privately and a NULL value would indicate a programming error, so a segfault is equally valid as a panic(), just less graceful.

Reply:
Oh I see, it's a different path than what used to have the default argument value.